desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphTransformers.java

```diff
@@ -235,7 +240,20 @@ public Icon transform(Object obj) {
                 } else {
                     configured = componentBuilder.isConfigured(false);
                 }
-                return IconUtils.getDescriptorIcon(descriptor, configured, IconUtils.ICON_SIZE_LARGE);
+
+                Icon descriptorIcon = IconUtils.getDescriptorIcon(descriptor, configured, IconUtils.ICON_SIZE_LARGE);
+
+                if (descriptor instanceof RemoteTransformerDescriptorImpl) {
```

Comment:
Since we are talking about an API and an interface here, I wouldn't want to restrict this to only remote transformers. The descriptor state stuff is capable of working on _all_ components, so let's code against that API ourselves too.